src/lib/transition_frontier/catchup_hash_tree.ml

```diff
@@ -106,7 +106,15 @@ let check_for_parent t h ~parent:p ~check_has_breadcrumb =
           ; ("tree", to_yojson t) ]
         "hash tree invariant broken: $parent not found in $tree for $hash"
   | Some x ->
-      if check_has_breadcrumb && x.state <> Have_breadcrumb then
+      if
+        check_has_breadcrumb
+        &&
+        match x.state with
```

Comment:
Can we pull this match out into another expression instead of nesting it into the `if`? eg `let cond = match ... with ... in if check_has_breadcrumb && cond then ... else ...`.

Reply:
I've replaced this with a call to an equality function, which ended up more readable than pulling out the condition.